nasa/earthdata-search
# EDSC-4234: Fix Swodlr form to include granule name, add guards to database writes, fix swodlr data encoding

# Overview

### What is the feature?
- Adds the name of the granule-id as part of the re-design but, aligns more with the original design spec
- fixes an issue where if you submitted a swodlr order then hit back and then forward again the resubmitted order failed because we were not encoding the swodlr data into the url
- Adds swodlr to outgoing metrics
- Adds an additional guard for retrieval downloads on swodlr orders > 10 granules
- Fixes an intermittent issue where the collectionMetadata in the Access method had undefined granule-id (This existed prior to the Access Method rewrite) the issue is that this is a child component of that using it (the only one) and needs to only render after that information is contained (it is not `null` though)


### What is the Solution?

Summarize what you changed.

### What areas of the application does this impact?

The swodlr form, access methods radio, the url encoding for collections

# Testing

### Reproduction steps

- **Environment for testing:*Prod*
- **Collection to test with:*C2799438271-POCLOUD*

1. Choose granules from this collection ensure you can download using the swodlr method
2. Ensure the new form includes the granule names
3. Ensure that refreshing the project page a few times works as expected (does not cause an error)
4. Ensure that submitting an order then hit the back button on the browser then download again (without changing anything) does not result in a failed download

### Attachments

3 granules: 
![image](https://github.com/user-attachments/assets/b038cd4e-e393-4e09-aed2-f0a784c9cf76)


Too many granules (10) 
<img width="1112" alt="Pasted Graphic 1" src="https://github.com/user-attachments/assets/7a785544-40b1-4662-b9e0-f14b7a35fa69">

Max granule amount:
<img width="1388" alt="image" src="https://github.com/user-attachments/assets/deafee19-f0b9-4ab1-9d77-ab347876fc8e">

Radio button contains external link to swodlr documentation within the tool-tip
<img width="1103" alt="image" src="https://github.com/user-attachments/assets/451b145c-6617-4d1d-8a08-6ef12f9b83d9">


# Checklist
